localstack/services/awslambda/provider.py

```diff
@@ -1766,13 +1806,27 @@ def remove_permission(
             raise ResourceNotFoundException(
                 f"Statement {statement_id} is not found in resource policy.", Type="User"
             )
-        if revision_id and function_permission.revision_id != revision_id:
+        fn_revision_id = self._function_revision_id(resolved_fn, resolved_qualifier)
+        if revision_id and revision_id != fn_revision_id:
             raise PreconditionFailedException(
-                "The Revision Id provided does not match the latest Revision Id. Call the GetFunction/GetAlias API to retrieve the latest Revision Id",
+                "The Revision Id provided does not match the latest Revision Id. "
+                "Call the GetFunction/GetAlias API to retrieve the latest Revision Id",
                 Type="User",
             )
         function_permission.policy.Statement.remove(statement)
-        function_permission.revision_id = FunctionResourcePolicy.new_revision_id()
+
+        # Update revision id for alias or version
+        # TODO: re-evaluate data model to prevent this dirty hack just for bumping the revision id
+        # TODO: does that need a `with function.lock` for atomic updates of the policy + revision_id?
+        if api_utils.qualifier_is_alias(resolved_qualifier):
+            latest_alias = resolved_fn.aliases[resolved_qualifier]
+            resolved_fn.aliases[resolved_qualifier] = dataclasses.replace(latest_alias)
+        # Assumes that a non-alias is a version
+        else:
+            latest_version = resolved_fn.versions[resolved_qualifier]
```

Comment:
For me the name `latest_version` is kind of confusing, as I would assume it was the `$LATEST` version, but as far as I can see, it should be whatever version is specified by the qualifier.
Not critical though, I see you just merged.

Reply:
oh right, good catch 👀  Thank you 🙏 
can fix it alongside the next PR